File: react/src/main/java/com/react/auth/model/RefreshToken.java
package com.react.auth.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "refresh_tokens")
@Getter
@Setter
@NoArgsConstructor()
@AllArgsConstructor
public class RefreshToken {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 255)
    private String token;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "expiry_at", nullable = false)
    private Instant expiryAt;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;
}

File: BaseSpringSecurity/src/main/kotlin/entity/RefreshToken.kt
package com.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "refresh_tokens")
open class RefreshToken(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    open val id: Int? = null,

    @Column(nullable = false, unique = true, length = 255)
    open val token: String,

    @Column(name = "user_id", nullable = false)
    open val userId: Int,

    @Column(name = "expiry_at", nullable = false)
    open val expiryAt: Instant,

    @Column(name = "created_at", nullable = false)
    open val createdAt: Instant = Instant.now()
)  {
    constructor() : this(null, "", 0, Instant.now(), Instant.now())
}
